### PR TITLE
RESOURCE-425 AWS::ApiGatewayV2::IntegrationResponse

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

1. It contains all the singular and plural resources.
2. Added the resource_id in the singular resource.
3. Modified the docs according to the new format. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill a box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — APPROVED** [review]

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+If this property is not defined, the response payload will be passed through from the integration response to the route response or method response without modification.
+
+`integration_response_id`
+: The integration response ID.
+
+`integration_response_key`
+: The integration response key.
+
+`response_parameters`
+: A key-value map specifying response parameters that are passed to the method response from the backend. The key is a method response header parameter name and the mapped value is an integration response header value, a static value enclosed within a pair of single quotes, or a JSON expression from the integration response body. The mapping key must match the pattern of method.response.header.{name}, where name is a valid and unique header name. The mapped non-static value must match the pattern of integration.response.header.{name} or integration.response.body.{JSON-expression}, where name is a valid and unique response header name and JSON-expression is a valid JSON expression without the $ prefix.
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md:57]: doesnt the name and json expression in the statement, "where {name} is a valid and unique response header name and {JSON-expression} is a valid JSON expression without the $ prefix." - to be covered within curly ({}) brackets?

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+`response_templates`
+: The collection of response templates for the integration response as a string-to-string map of key-value pairs. Response templates are represented as a key/value map, with a content-type as the key and a template as the value.
+
+`template_selection_expression`
+: The template selection expressions for the integration response.
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md:63]: Added sentence - Supported only for WebSocket APIs. Is this fine?

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+: **Field**: `integration_response_id`
+
+`integration_response_keys`
+: The integration response key.
+
+: **Field**: `integration_response_key`
+
+`response_parameters`
+: A key-value map specifying response parameters that are passed to the method response from the backend. The key is a method response header parameter name and the mapped value is an integration response header value, a static value enclosed within a pair of single quotes, or a JSON expression from the integration response body. The mapping key must match the pattern of method.response.header.{name}, where name is a valid and unique header name. The mapped non-static value must match the pattern of integration.response.header.{name} or integration.response.body.{JSON-expression}, where name is a valid and unique response header name and JSON-expression is a valid JSON expression without the $ prefix.
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md:60]: Here too the name and jsonexpression needs to be curly braces?

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+: **Field**: `response_parameters`
+
+`response_templates`
+: The collection of response templates for the integration response as a string-to-string map of key-value pairs. Response templates are represented as a key/value map, with a content-type as the key and a template as the value.
+
+: **Field**: `response_templates`
+
+`template_selection_expressions`
+: The template selection expressions for the integration response.
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md:70]: 2nd sentence added is fine?

**Review by dkumaras — COMMENTED** [review]
Pls check the changes and confirm.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md (hunk 1 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md]: Not required

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md (hunk 2 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md]: Yes

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md (hunk 3 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md]: This is fine

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md (hunk 4 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md]: Yes

**Comment by soumyo13** [comment]
> Pls check the changes and confirm.

Replied to your comments.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md (hunk 1 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md]: ok

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md (hunk 2 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_response.md]: ok

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md (hunk 3 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md]: ok

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md (hunk 4 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration_responses.md]: ok

**Review by dkumaras — APPROVED** [review]

**Pull request merged**